ios/RNSScreen.m

```diff
@@ -142,6 +149,50 @@ - (void)setStackAnimation:(RNSScreenStackAnimation)stackAnimation
   }
 }
 
+- (void)setStatusBarStyle:(RNSStatusBarStyle)statusBarStyle
+{
+  if (statusBarStyle != _statusBarStyle) {
+    [RNSScreenView checkUIVCBasedStatusBarAppearance];
+    _statusBarStyle = statusBarStyle;
+    _controller.modalPresentationCapturesStatusBarAppearance = YES;
+    [UIView animateWithDuration:0.5 animations:^{
+      [self->_controller setNeedsStatusBarAppearanceUpdate];
+    }];
+  }
+}
+
+- (void)setStatusBarAnimation:(UIStatusBarAnimation)statusBarAnimation
+{
+  if (statusBarAnimation != _statusBarAnimation) {
+    [RNSScreenView checkUIVCBasedStatusBarAppearance];
+    _statusBarAnimation = statusBarAnimation;
+    _controller.modalPresentationCapturesStatusBarAppearance = YES;
+    [UIView animateWithDuration:0.5 animations:^{
+      [self->_controller setNeedsStatusBarAppearanceUpdate];
+    }];
+  }
+}
+
+- (void)setStatusBarHidden:(BOOL)statusBarHidden
+{
+  if (statusBarHidden != _statusBarHidden) {
+    [RNSScreenView checkUIVCBasedStatusBarAppearance];
+    _statusBarHidden = statusBarHidden;
+    _controller.modalPresentationCapturesStatusBarAppearance = YES;
+    [UIView animateWithDuration:0.5 animations:^{
+      [self->_controller setNeedsStatusBarAppearanceUpdate];
+    }];
+  }
+}
+
++ (void)checkUIVCBasedStatusBarAppearance
+{
+  if (viewControllerStatusBarDisabled) {
```

Comment:
there is no need to define variables outside or mark them as extern. I'd perhaps rename this method to `assertViewControllerBasedStatusBarAppearenceSet` then the implementation should look as follows:


```objc
+ (void)assertViewControllerBasedStatusBarAppearenceSet
{
    static dispatch_once_t once;
    static bool viewControllerBasedAppearence;
    dispatch_once(&once, ^{
viewControllerBasedAppearence = [[[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"] boolValue];
    });
    if (!viewControllerBasedAppearence) {
      RCTLogError(@"........");
    }
}
```